zerver/actions/realm_settings.py

```diff
@@ -208,6 +210,19 @@ def parse_and_set_setting_value_if_required(
     return parsed_value, setting_value_changed
 
 
+def validate_authentication_methods_dict_from_api(
+    realm: Realm, authentication_methods: Dict[str, bool]
+) -> None:
+    current_authentication_methods = realm.authentication_methods_dict()
+    for name in authentication_methods:
+        if name not in current_authentication_methods:
+            raise JsonableError(
+                _("Invalid authentication method: {name}. Valid methods are: {methods}").format(
+                    name=name, methods=sorted(current_authentication_methods.keys())
+                )
+            )
```

Comment:
An alternative approach would be to just silently drop any such keys from the dict and proceed - instead of throwing an error. But I think this endpoint is only used by the webapp, so we can set the expectation of only receiving keys that match backends currently enabled on the server.

(If you don't enable e.g. GitHub auth in `settings.py`, then it doesn't show up in the `Authentication methods` menu in `Organization settings` and `GitHub` key isn't in the dict submitted to the API by the webapp)